core/src/main/java/kafka/server/AssignmentsManager.java

```diff
@@ -210,6 +220,9 @@ public void run() throws Exception {
             channelManager.sendRequest(new AssignReplicasToDirsRequest.Builder(
                     buildRequestData(brokerId, brokerEpochSupplier.get(), assignment)),
                     new AssignReplicasToDirsRequestCompletionHandler());
+            inflight.values().stream()
+                    .filter(assignmentEvent -> assignmentEvent.callback != null)
+                    .forEach(assignmentEvent -> assignmentEvent.callback.accept(DirectoryEventRequestState.DISPATCHED));
```

Comment:
I don't think we need the state in the callback. It makes sense to distinguish between the 3 states, but I don't see why we callback should fire for anything but `COMPLETED`.

Reply:
For dispatched this is true but for queued I think we still need this so we don't send the reassignment multiple times to AssignmentManager. Especially that as far as I see the manager isn't idempotent.